atecc608a/main.c

```diff
@@ -194,180 +208,98 @@ psa_status_t test_psa_import_verify()
         0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
         0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
     };
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+    ASSERT_SUCCESS_PSA(psa_open_key(atecc608a_private_key_slot,
+                                    private_handle));
 
-    ASSERT_SUCCESS_PSA(atecc608a_drv_info.p_asym->p_sign(
-                           atecc608a_private_key_slot, alg, hash, sizeof(hash),
-                           signature, sizeof(signature), &signature_length));
+    ASSERT_SUCCESS_PSA(psa_asymmetric_sign(*private_handle, alg, hash, sizeof(hash),
+                                           signature, sizeof(signature),
+                                           &signature_length));
+
+    ASSERT_SUCCESS_PSA(psa_export_public_key(*private_handle, pubkey,
+                                             sizeof(pubkey), &pubkey_len));
 
-    ASSERT_SUCCESS_PSA(atecc608a_drv_info.p_key_management->p_export(
-                           atecc608a_private_key_slot, pubkey, sizeof(pubkey),
-                           &pubkey_len));
     /*
      * Import the secure element's public key into a volatile key slot.
      */
-    ASSERT_SUCCESS_PSA(psa_allocate_key(&verify_handle));
 
-    psa_key_policy_set_usage(&policy, PSA_KEY_USAGE_VERIFY, alg);
-    ASSERT_SUCCESS_PSA(psa_set_key_policy(verify_handle, &policy));
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_VERIFY);
+    psa_set_key_algorithm(&attributes, alg);
+    psa_set_key_type(&attributes, key_type);
 
-    ASSERT_SUCCESS_PSA(psa_import_key(verify_handle, key_type, pubkey,
-                                      pubkey_len));
+    ASSERT_SUCCESS_PSA(psa_import_key(&attributes, pubkey,
+                                      pubkey_len, &verify_handle));
 
     /* Verify that the signature produced by the secure element is valid. */
     ASSERT_SUCCESS_PSA(psa_asymmetric_verify(verify_handle, alg, hash,
                                              sizeof(hash), signature,
                                              signature_length));
 
+    psa_close_key(*private_handle);
+    psa_close_key(verify_handle);
     printf("test_psa_import_verify succesful!\n");
 exit:
     return status;
 }
 
-/* Test that a public key generated while generating a private key can
- * be imported. */
-psa_status_t test_generate_import()
```

Comment:
Why can't we test that a public key generated while generating a private key can be imported anymore?

Reply:
Last time I checked, the psa key generation function discarded the returned public key, as it would require implementing a storage. Not sure if it's the same in the current state though.

Reply:
Can we call export later instead of using the public key returned by the driver's generate?

Reply:
Yes, we're performing such tests.